lib/perl/Genome/InstrumentData/Command/Import/CsvParser.pm

```diff
@@ -10,57 +10,58 @@ require List::MoreUtils;
 use Params::Validate qw( :types );
 use Text::CSV;
 
-class Genome::InstrumentData::Command::Import::GenerateBase {
-    is => 'Command::V2',
-    is_abstract => 1,
-    has_input => {
+class Genome::InstrumentData::Command::Import::CsvParser {
+    is => 'UR::Object',
+    has => {
         file => {
             is => 'Text',
             doc => 'Comma (.csv) or tab (.tsv) separated file of entity names, attributes and other meta data. Separator is determined by file extension.',
         },
     },
-    has_optional_output => {
-        output_file => {
-            is => 'Text',
-            default_value => '-',
-            doc => 'Output file to put the commands to create the needed entities for import.',
-        },
-    },
     has_optional_transient => {
-        _output => { is => 'ARRAY', default_value => [], },
-        _instdata_property_names => { is => 'ARRAY', },
         _entity_attributes => { is => 'ARRAY', },
-        _nomenclature => { is => 'Text', },
-        _names_seen => { is => 'HASH', default_value => {}, },
+        _fh => { },
+        _parser => { },
     }
 };
 
-sub help_detail {
+sub csv_help {
     return <<HELP;
-The file should be a comma or tab separated values and indicated with the appropriate extension (csv and tsv). Column headers to use to generate the create commands should start with the entity (individual, sample, library, instdata) name then a period (.) and then then attribute name (Ex: sample.name_part). Here are some required and optional columns. For more, see each entity's create command (Ex: genome sample create --h). Please see Confluence documentation for more information and a full example.
+This is a comma (.csv) or tab (.tsv) separated file of entity names, attributes and other meta data. Separator is determined by file extension. Column headers to use to generate the create commands should start with the entity (individual, sample, library, instdata) name then a period (.) and then then attribute name (Ex: sample.name_part). Here are some required and optional columns. For more, see each entity's create command (Ex: genome sample create --h). Please see Confluence documentation for more information and a full example.
```

Comment:
* The previous command referred to `metadata` but this refers to `meta data`.  These should be consistent.
* The sentence beginning `Column headers...` is a bit hard to follow.  Maybe some additional punctuation or a rewording?  One possibility: `Column headers should start with the entity (individual, sample, library, instdata) name followed by a period (.) and the attribute name (Ex: sample.name_part).`
* I recommend including `--help` in the documentation over `--h`. Alternatively `-h` would be sufficient.